pkg/controllers/cluster/direct_access_controller.go

```diff
@@ -160,3 +179,26 @@ func generateNewClientKubeConfig(_ context.Context, restConfigGetter *clientutil
 	}
 	return kubeconfigByte, nil
 }
+
+func deletePlugins(ctx context.Context, c client.Client, cluster *greenhousev1alpha1.Cluster) (count int, err error) {
+	pluginList := &greenhousev1alpha1.PluginList{}
+	err = c.List(
+		ctx,
+		pluginList,
+		client.InNamespace(cluster.GetNamespace()),
+		client.MatchingLabels{greenhouseapis.LabelKeyCluster: cluster.GetName()},
+	)
+	if err != nil {
+		return
+	}
+	for _, plugin := range pluginList.Items {
+		plugin.SetAnnotations(map[string]string{
+			"greenhouse.sap/deletedAt": strconv.FormatInt(time.Now().Unix(), 10),
```

Comment:
This looks new. Why not put it into [well_known.go](https://github.com/cloudoperators/greenhouse/blob/main/pkg/apis/well_known.go) and make it a best practice in future development/refactorings to annotate resources we delete?

Reply:
this doesn't matter, i have removed it.